src/GTP.cpp

```diff
@@ -516,18 +516,30 @@ bool GTP::execute(GameState & game, std::string xinput) {
     } else if (command.find("heatmap") == 0) {
         std::istringstream cmdstream(command);
         std::string tmp;
-        int rotation;
+        std::string rotation;
+        Network::Netresult vec;
 
         cmdstream >> tmp;   // eat heatmap
         cmdstream >> rotation;
 
-        if (!cmdstream.fail()) {
-            auto vec = Network::get_scored_moves(
-                &game, Network::Ensemble::DIRECT, rotation, true);
-            Network::show_heatmap(&game, vec, false);
+        if (!cmdstream.fail() && rotation == "all") {
+            for (int r=0; r<8; r++) {
+                vec = Network::get_scored_moves(
+                    &game, Network::Ensemble::DIRECT, r, true);
+                Network::show_heatmap(&game, vec, false);
+            }
         } else {
-            auto vec = Network::get_scored_moves(
-                &game, Network::Ensemble::DIRECT, 0, true);
+            if (cmdstream.fail()) {
+                vec = Network::get_scored_moves(
```

Comment:
Just set rotation = 0 and merge this with the else branch?

Reply:
Done